Libraries/LibGUI/TextBox.h

```diff
@@ -35,6 +35,27 @@ class TextBox : public TextEditor {
 public:
     TextBox();
     virtual ~TextBox() override;
+
+    Function<void()> on_up_pressed;
+    Function<void()> on_down_pressed;
+
+    void set_history_enabled(bool enabled) { is_history_enabled = enabled; }
+    void add_current_text_to_history();
+
+private:
+    virtual void keydown_event(GUI::KeyEvent&) override;
+
+    bool has_any_history() const { return !m_input_history.is_empty(); }
+    bool has_previous_input() const { return m_current_historical_input > 1; }
+    bool has_next_input() const { return m_current_historical_input < m_input_history.size(); }
```

Comment:
These look a bit strange; I would expect to "have previous input" if I'm at index 1.
Also, "previous input" is rather ambiguous, can we call these something like `can_go_backwards_in_history` / `can_go_forwards_in_history` to make it a bit clearer what's going on?

Reply:
Yeah, renaming them sounds good. They're off set like that because I was making this a whole lot more complicated then it needed to be, let me make them 0-indexed again :^)